### DISCUSSION TITLE
Can you connect to bluetooth speaker via mac address

### DISCUSSION THREAD

**Discussion by SorabhR** [pschatzmann/ESP32-A2DP#595] [Q&A] [answered]
### Problem Description

Hey,

in a2dp source, while connecting a bluetooth speaker if there are multiple with same name, is there a way to connect via mac address of the speaker.

### Device Description

ESP32 development board

### Sketch

```cpp
a2dp_send example
```

### Provide your Version of the EP32 Arduino Core (or the IDF Version)

3.0.0

### I have checked existing issues, discussions and online documentation

- [X] I confirm I have checked existing issues, discussions and online documentation

**Comment by pschatzmann** [comment]
I don't understand why you think this is a bug. 
So I am converting this to a discussion...

**Comment by pschatzmann** [comment, marked as answer]
That's what the set_ssid_callback callback and example is about!

**Comment by spiralbrain** [comment]
Hi I am on the same page here.

The problem arises when we have multiple devices with the same SSID. What I have observed is that if we connect to one SSID the MAC binds to the Bluetooth receiver. Then if we try to pair another device with same SSID it cannot pair until the ESP32 is erased completely and reflashed. Then there is also an issue where the ESP32 where it stores previous SSID's and MACS. If without doing a complete erase a new device is paired, The previous devices are remembered. We may need to dig deeper into the SDK and hardware implementation to see how this happens. I feel if there could be a way to just pair using MAC's ignoring the SSID it would be of a great help.

Thanks Phil for your hard work!
